### PR TITLE
Update @types/webscopeio__react-textarea-autocomplete to match 4.7.3 …

### PR DESCRIPTION
**Pull request by tjsilver**
…of [@webscopeio/react-textarea-autocomplete](https://github.com/webscopeio/react-textarea-autocomplete)

Co-authored-by: @twrichards

Updated type defininitions to match https://github.com/webscopeio/react-textarea-autocomplete/tree/v4.7.3#props.

- Added missing props for:
  - `boundariesElement`
  - `textAreaComponent` (including new type parameter `CustomTextAreaProps` to make the use of custom textarea components typesafe, type parameter defaults to the standard props for `<textarea>`s
  - `renderToBody`
  - `onItemHighlighted`
  - `onItemSelected`.  
- Removed old prop `closeOnClickOutside`.
- Changed order of props to match order in documentation.
- Replaced  uses of `React.SFC` (because it's deprecated) with a broader type definition which supports both functional and class components.
- Corrected the shape of `loadingComponent` since it receives `data` as a prop.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  https://github.com/webscopeio/react-textarea-autocomplete/tree/v4.7.3#props
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/webscopeio__react-textarea-autocomplete/index.d.ts
Comment (michauzo) [inline types/webscopeio__react-textarea-autocomplete/index.d.ts:107]: Long line. Please break it.

**Review by michauzo — CHANGES_REQUESTED** [review]
Overall this looks good. Thanks for contributing. I've left one cosmetic comment. I wasn't sure if you can make corrections after I choose approve, so I chose Request changes. I'll approve it once this is corrected. Thanks again.

**Review by michauzo — APPROVED** [review]

**Comment by twrichards** [comment]
Ready to merge

**Comment by tjsilver** [comment]
ready to merge

**Pull request merged**
